exp/controllers/machinepool_controller_phases.go

```diff
@@ -97,6 +97,7 @@ func (r *MachinePoolReconciler) reconcileExternal(ctx context.Context, cluster *
 	}
 
 	// Initialize the patch helper.
+	// NOTE: This object doesn't implement observedGeneration. No need to add the WithStatusObservedGeneration option.
```

Comment:
This comment seems a bit misleading, we shouldn't be updating observedGeneration because we are not managing this resource, just trying to get information from an externally reconciled resource.